packages/algob/src/lib/asa.ts

```diff
@@ -5,25 +5,45 @@ import { BuilderError } from "../internal/core/errors";
 import { ERRORS } from "../internal/core/errors-list";
 import { ASSETS_DIR } from "../internal/core/project-structure";
 import { parseZodError } from "../internal/core/validation-errors";
-import { ASADefs } from "../types";
+import { Accounts, ASADef, ASADefs } from "../types";
 import { ASADefsSchema } from "../types-input";
 import { loadFromYamlFileSilentWithMessage } from "./files";
 
-export function validateASADefs (obj: Object, filename?: string): ASADefs {
+function validateSingle (accounts: Accounts, filename: string, asaDef: ASADef): void {
+  if (!asaDef.optInAccNames || asaDef.optInAccNames.length === 0) {
+    return;
+  }
+  for (const accName of asaDef.optInAccNames) {
+    if (!accounts.get(accName)) {
+      throw new BuilderError(
+        ERRORS.SCRIPT.ASA_PARAM_ERROR_NO_NAMED_OPT_IN_ACCOUNT, {
+          filename: filename,
+          optInAccName: accName
+        });
+    }
+  }
+}
+
+function validateASADefsOneByOne (accounts: Accounts, asaDefs: ASADefs, filename: string): void {
```

Comment:
We don't need to be that verbose, let's rename function  to `validateASADefs` (or feel free to propose alternative)